### PR TITLE
Feature/blacklist Ability to blacklist members for future charges

### PR DESCRIPTION
**Pull request by vonElfvin**

### CODE REVIEW

File: src/main/kotlin/com/hedvig/accountservice/schedulers/domain/commands/BlacklistCommand.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.hedvig.accountservice.schedulers.domain.commands
+
+import org.axonframework.commandhandling.TargetAggregateIdentifier
+
+data class BlacklistCommand(
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/accountservice/schedulers/domain/commands/BlacklistCommand.kt]: Shall we name this differently?

```suggestion
data class SetBlacklistStateCommand(
```

File: src/main/kotlin/com/hedvig/accountservice/schedulers/domain/events/BlacklistDeniedEvent.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.hedvig.accountservice.schedulers.domain.events
+
+import com.hedvig.accountservice.schedulers.model.SchedulerState
+
+data class BlacklistDeniedEvent(
+    val memberId: String,
+    val currentState: SchedulerState,
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/accountservice/schedulers/domain/events/BlacklistDeniedEvent.kt]: Do we need the currentState and the nextState? Isn't the Event explicit enough?

File: src/main/kotlin/com/hedvig/accountservice/schedulers/domain/events/BlacklistedEvent.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.hedvig.accountservice.schedulers.domain.events
+
+import com.hedvig.accountservice.schedulers.model.SchedulerState
+
+data class BlacklistedEvent(
+    val memberId: String,
+    val nextState: SchedulerState
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/accountservice/schedulers/domain/events/BlacklistedEvent.kt]: Same here, i guess we can figure out from the event what is the nextState

File: src/main/kotlin/com/hedvig/accountservice/schedulers/domain/events/BlacklistedEvent.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.hedvig.accountservice.schedulers.domain.events
+
+import com.hedvig.accountservice.schedulers.model.SchedulerState
+
+data class BlacklistedEvent(
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/accountservice/schedulers/domain/events/BlacklistedEvent.kt]:
```suggestion
data class BlacklistStateSetEvent(
```

File: src/main/kotlin/com/hedvig/accountservice/schedulers/web/ScheduleController.kt [hunk 5]
```diff
@@ -170,4 +170,16 @@ class ScheduleController @Autowired constructor(
         ))
         return ResponseEntity.noContent().build()
     }
+
+    @PostMapping("/blacklist/{memberId}")
+    fun blacklist(
+        @PathVariable memberId: String,
+        @RequestParam blacklistedBy: String
+    ): ResponseEntity<Void> {
+        commandGateway.sendAndWait<Void>(BlacklistCommand(
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/accountservice/schedulers/web/ScheduleController.kt]: Should we use the `SchedulerStateTransition` that we return from the commandHandler in order to signal if the change was accepted or not?

File: src/main/kotlin/com/hedvig/accountservice/schedulers/domain/events/BlacklistDeniedEvent.kt (hunk 2 above)
Comment (vonElfvin) [reply src/main/kotlin/com/hedvig/accountservice/schedulers/domain/events/BlacklistDeniedEvent.kt]: Depends on what we want I guess. With this we know what was attempted (i.e. we know from where to where by whom)

File: src/main/kotlin/com/hedvig/accountservice/schedulers/domain/events/BlacklistedEvent.kt (hunk 3 above)
Comment (vonElfvin) [reply src/main/kotlin/com/hedvig/accountservice/schedulers/domain/events/BlacklistedEvent.kt]: nextState is required to set the state in the @EventSourcingHandler and add to the database in the @EventHandler

File: src/main/kotlin/com/hedvig/accountservice/schedulers/domain/events/BlacklistedEvent.kt (hunk 4 above)
Comment (vonElfvin) [reply src/main/kotlin/com/hedvig/accountservice/schedulers/domain/events/BlacklistedEvent.kt]: Changed it to "SchedulerBlacklistedEvent"

File: src/main/kotlin/com/hedvig/accountservice/schedulers/domain/commands/BlacklistCommand.kt (hunk 1 above)
Comment (vonElfvin) [reply src/main/kotlin/com/hedvig/accountservice/schedulers/domain/commands/BlacklistCommand.kt]: Changed it to "BlacklistSchedulerCommand"

File: src/main/kotlin/com/hedvig/accountservice/schedulers/web/ScheduleController.kt (hunk 5 above)
Comment (vonElfvin) [reply src/main/kotlin/com/hedvig/accountservice/schedulers/web/ScheduleController.kt]: Good catch!

File: src/main/kotlin/com/hedvig/accountservice/schedulers/domain/commands/BlacklistCommand.kt (hunk 1 above)
Comment (meletios) [reply src/main/kotlin/com/hedvig/accountservice/schedulers/domain/commands/BlacklistCommand.kt]: 👍

**Comment by vonElfvin** [comment]
Replaced this PR with a rebased one

**Pull request closed without merging**